jerry-core/vm/vm.c

```diff
@@ -2010,10 +2001,37 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         {
           if (ecma_are_values_integer_numbers (left_value, right_value))
           {
-            ecma_integer_value_t left_integer = (ecma_integer_value_t) left_value;
-            ecma_integer_value_t right_integer = (ecma_integer_value_t) right_value;
+            bool is_less = (ecma_integer_value_t) left_value < (ecma_integer_value_t) right_value;
+
+            if (*byte_code_p <= CBC_BRANCH_IF_TRUE_BACKWARD_3 && *byte_code_p >= CBC_BRANCH_IF_TRUE_BACKWARD)
+            {
+              byte_code_start_p = byte_code_p++;
+              branch_offset_length = CBC_BRANCH_OFFSET_LENGTH (*byte_code_start_p);
+              JERRY_ASSERT (branch_offset_length >= 1 && branch_offset_length <= 3);
+
+              branch_offset = *(byte_code_p++);
+
+              if (unlikely (branch_offset_length != 1))
+              {
+                branch_offset <<= 8;
+                branch_offset |= *(byte_code_p++);
+                if (unlikely (branch_offset_length == 3))
+                {
+                  branch_offset <<= 8;
+                  branch_offset |= *(byte_code_p++);
+                }
+              }
+
+              if (is_less)
+              {
+                /* Note: The opcode is a bacward branch. */
+                byte_code_p = byte_code_start_p - branch_offset;
```

Comment:
btw, if `is_less` is false, did we do the `branch_offset` computation above in vain? if `!is_less`, wouldn't be `byte_code_p += branch_offset_length` enough?

Reply:
yes, thanks! updated.